<launch>

	<param
		name="/use_sim_time"
		value="true" />

	<param
		name="robot_description"
		command="$(find xacro)/xacro.py '$(find lrm_description)/urdf/clubcar/robot.xacro'" />
<!-- 
	<node
		name="robot_state_publisher"
		pkg="robot_state_publisher"
		type="robot_state_publisher"
		output="screen"
		respawn="true"
	>
		<param
			name="publish_frequency"
			value="30" />
	</node>

	<node
		name="joint_state_publisher"
		pkg="joint_state_publisher"
		type="joint_state_publisher" />
-->
	<node
		pkg="tf"
		type="static_transform_publisher"
		name="odom_to_map"
		args="0 0 0 0 0 0 world odom 100" />

	<node
		name="gps_odom"
		pkg="lrm_odometry"
		type="utm_odometry_node"
		output="screen"
	>
		<remap
			from="odom"
			to="/odom_gps" />
		<remap
			from="fix"
			to="/lse_xsens_mti/xsens/fix" />
		<param
			name="rot_cov"
			value="99999" /> <!-- uncertain -->
		<param
			name="pos_cov"
			value="25" /> <!-- err 5m -->
		<param
			name="frame_id"
			value="/odom" />
		<param
			name="child_frame_id"
			value="base_odometry" />
		<param
			name="latitude"
			value="-22.006286" />
		<param
			name="longitude"
			value="-47.898417" />
	</node>

	<node
		pkg="robot_pose_ekf"
		type="robot_pose_ekf"
		name="robot_pose_ekf"
		output="screen"
	>
		<remap
			from="imu"
			to="/lse_xsens_mti/xsens/imu/data" />
		<remap
			from="odom"
			to="/odom" />
		<remap
			from="vo"
			to="/odom_gps" />

		<param
			name="output_frame"
			value="odom" />
		<param
			name="freq"
			value="15.0" />
		<param
			name="sensor_timeout"
			value="0.5" />
		<param
			name="publish_tf"
			value="true" />
		<param
			name="odom_used"
			value="true" />
		<param
			name="imu_used"
			value="true" />
		<param
			name="vo_used"
			value="true" />
	</node>

	<node
		pkg="lrm_odometry"
		type="simple_odometry"
		name="imu_odometry"
		output="screen"
	>
		<param
			name="rate"
			value="15" />
		<param
			name="print_info"
			value="false" />
		<param
			name="publish_tf"
			value="false" />
		<param
			name="publish_js"
			value="false" />
		<param
			name="frame_id"
			value="odom" />
		<remap
			from="imu_data"
			to="/lse_xsens_mti/xsens/imu/data" />
	</node>

	<node
		name="rviz"
		pkg="rviz"
		type="rviz"
		respawn="true"
		output="screen" />
 
</launch>